Gemfile

```diff
@@ -1,2 +1,5 @@
 source "https://www.rubygems.org"
 gemspec
+
+gem 'daru', :path => "/home/sameer/gitrepos/daru/"
+gem 'gsl-nmatrix', :path => "/home/sameer/gitrepos/gsl-nmatrix/"
```

Comment:
Another thing to have in mind is that, according to your proposed timeline, this pull request should be finished & merged by the end of the 3rd week of the coding period (which only starts next week, officially). Thus it's totally OK to have this line for testing right now. However, keep in mind that I don't want to introduce GSL as a dependency. :P

All in all, you're doing a good job. I'll finish reading your commits today as time permits. I'll put notes in the code itself or email them to you. ciao! :+1: